### ISSUE TITLE
Ctrl-d in REPL sometimes does nothing

### ISSUE THREAD

**Issue by lendle** [JuliaLang/julia#7605]
I've found that if I evaluate an expression in the REPL and hit ctrl-D before the evaluation completes, ctrl-D does nothing even after the `julia>` prompt back until I hit enter.

E.g.:

``` jlcon
julia> whos()
Base                          Module
Core                          Module #hit ctrl-D some time before whos() completes
Main                          Module
ans                           Nothing

julia> #hit ctrl-D here one or more times, does nothing, hit enter

julia> #now ctrl-D works, julia exits
```

This is not a huge issue, but may be related to a segfault I got and cannot reproduce. Specifically, I did `Pkg.checkout("Optim")` and hit ctrl-d either while that was executing or immediately after it completed. Unfortunately I cannot remember but I THINK it was while checkout was executing.

**Comment by thepulkitagarwal** [comment]
On Ubuntu 14.04, this also happens when you type anything (`julia> abc`) and then do ctrl+D.

Ctrl+C works. And Ctrl+U followed by Ctrl+D also works.

**Comment by rfourquet** [comment]
That seems fixed? (maybe by #21505?)

**Comment by mbauman** [comment]
Can confirm on a Linux machine wherein 0.5 has this problem but it's fixed on 0.6 and master.
